### ISSUE TITLE
Crash on operation: PromiseStatus on iOS but not on macOS

### ISSUE THREAD

**Issue by stefanhp** [malcommac/Hydra#69]
We are using Hydra as part of a client API framework. There is a version without `Promise` that uses `Alamofire` for networking. We then add a `Promise` based wrapper for each API call.
The framework exists for both iOS and macOS.
Recently we upgraded to Xcode 11.2, swift 5.1 and Hydra 2.0.2 (from 1.2.1, using SPM).

Now our tests for macOS pass, but the iOS tests crash in Promise.swift, line 71:
```swift
	 public lazy var operation: PromiseStatus = {
		return PromiseStatus(token: self.invalidationToken, { [weak self] () -> () in
			self?.set(state: .cancelled)
		})
	}()
```

I tried to debug but could not figure out what is going on.

Here is [a sample project](https://github.com/Myotest/hydra-issue-69-macOS-iOS) that reproduces the bug.

The project has two targets, one for iOS and one for macOS. Both use the same source and test swift files.
The macOS tests pass but the iOS promise test fails (the standard api test passes, so the api part is ok).

**Comment by oabramov** [comment]
It looks like this problem is faced by all developers using SPM -
https://twitter.com/tcldr1/status/1190396639334285312?s=21For me the workaround with using the build setting DEAD_CODE_STRIPPING = NO are worked.

**Comment by stefanhp** [comment]
I confirm that this is fixed by Xcode 11.3 / Swift 5.1.3.
